File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n = nums.length;
        int sum =0;
        for(int i =0;i<k ; i++ ){
            sum+= nums[i];
        }
        int maxsum  = sum;
        int low = 0;
        int high = k;
        while(high<n){
            sum = sum - nums[low];
            sum = sum+ nums[high];
            
            maxsum = Math.max(maxsum,sum);
            low++;
            high++;
        }
        return (double) maxsum / k;
        
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: list[int], k: int) -> float:
        low =0
        high=k
        total =0
        for i in range(k):
            total+=nums[i]
        maxsum = total
        while(high<len(nums)):
            total = total+nums[high]-nums[low]
            maxsum = max(maxsum,total)
            low+=1
            high+=1

        return maxsum/k

        